### DISCUSSION TITLE
Fetch/XHR Call Within Budibase Javascript?

### DISCUSSION THREAD

**Discussion by CosmasSC128** [Budibase/budibase#4763] [Help] [answered]
Please someone let me know if this is possible. I have tried to use any kind of fetch and they are all cancelled within a few ms. Is there a header / security configuration or something I'm missing?

It seems javascript 'onload' is also blocked? Thank you for any help that can be provided.

**Comment by mjashanks** [comment, marked as answer]
Thats true. The client-side javascript is meant only for data manipulations. We have blocked access to the DOM and to fetch.

The way to achieve this is to
1. Setup a REST API datasource
2. Use your datasource either by 
- Adding a `Data Provider` to your page, which connects to it. Used for fetching (GET) data
- On Button Click, use the `Execute Query` action to make POST requests

Does this cover what you need?

**Reply by CosmasSC128** [reply]
It doesn't. Essentially my issue was having to fetch an image blob from the bubibase local storage of an image that had been dropped into the image uploader, so that I could translate it into a base64 URL and post that somewhere. But, seems like there's another work around since there's no way to get an image blob out of a budibase local URL.

Unless I'm missing something else. Thanks for the response either way.

**Reply by mjashanks** [reply]
Ah yea - Budibase file handling is in need of some improvement overall. I don't think this is going to possible right now. The way I would envision this in future is for attachments to be handled better via automations. I have no solution for you today though, sorry.

**Reply by CosmasSC128** [reply]
Thank you for the responses. It's already a wonderful product, I'm sure it will only improve with time.

**Comment by yatharth-zeotap** [comment]
Hi @mjashanks 

> Thats true. The client-side javascript is meant only for data manipulations. We have blocked access to the DOM and to fetch.


I am facing a similar issue. I think recently this was modified, 
cause earlier I was able to make a get call using this custom JS option.
I want to fill a "Text" field with the REST API response, how do I do this, creating a rest source might not be able to fulfill this.

This is the error I saw in the console.
<img width="551" alt="image" src="https://user-images.githubusercontent.com/74049974/157391983-0777dbdc-334a-48c0-91c2-f48e78ef349d.png">

**Reply by mjashanks** [reply]
@yatharth-zeotap  you should be able to achieve this using a REST data source....

1. Add your REST data source and query
2. In `Design`, surround your form with a `Data Provider`, connected to your REST query
3. Now, you can set the default value for your text input to something like `{{ Your Data Provider.Rows.0.ValueFromApi }}`
